File: app/src/main/java/io/github/wulkanowy/di/AppModule.java
package io.github.wulkanowy.di;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Binds;
import dagger.Module;
import io.github.wulkanowy.WulkanowyApp;
import io.github.wulkanowy.data.Repository;
import io.github.wulkanowy.data.RepositoryContract;

@Module
public abstract class AppModule {

    @Binds
    abstract Context provideContext(WulkanowyApp app);

    @Singleton
    @Binds
    abstract RepositoryContract provideRepository(Repository repository);
}

File: app/src/main/java/io/github/wulkanowy/di/AppModule.kt
package io.github.wulkanowy.di

import android.content.Context
import dagger.Module
import dagger.Provides
import io.github.wulkanowy.WulkanowyApp
import io.github.wulkanowy.utils.schedulers.AppSchedulers
import io.github.wulkanowy.utils.schedulers.SchedulersManager
import io.reactivex.disposables.CompositeDisposable

@Module
internal class AppModule {

    @Provides
    fun provideContext(app: WulkanowyApp): Context = app

    @Provides
    fun provideCompositeDisposable(): CompositeDisposable = CompositeDisposable()

    @Provides
    fun provideSchedulers(): SchedulersManager = AppSchedulers()
}
